### PR TITLE
Store DMN decision in the state

### LINKED ISSUE #8069: Store DMN decision in the state
**Description**

We should store the DMN decisions and DRGs in the RocksDB state and allow access to the deployed decisions. We need the access to determine the version when deploying a new DMN resource and to evaluate a decision. 

* create a new state for DMN decisions and DRGs
  * the mutable interface allows inserting a new decision and a new DRG based on the records
  * the immutable interface allows accessing the latest decision/DRG based on a given decision id
  * cache the transformed/parsed decisions/DRGs to avoid costly parsing on repeated access
* create new event appliers for the decision/DRG `CREATED` event
  * insert the decision/DRG in the state         

blocked by #8067

### PR DESCRIPTION
**Pull request by saig0**
## Description

* create a new state to store DMN decisions and DRGs
  * the state returns a copy of the persisted object to avoid any side effects of reusing the same object
  * check the version of the decision/DRG before updating the latest version to avoid inconsistencies (i.e. defensive programming)    
* put the DMN decisions and DRGs in the state when applying the CREATED events

## Related issues

closes #8069 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java [hunk 1]
```diff
@@ -62,6 +64,11 @@ public EventAppliers(final MutableZeebeState state) {
     registerProcessMessageSubscriptionEventAppliers(state);
     registerTimeEventAppliers(state);
     registerProcessEventAppliers(state);
+
+    register(DecisionIntent.CREATED, new DecisionCreatedApplier(state.getDmnDecisionState()));
+    register(
+        DecisionRequirementsIntent.CREATED,
+        new DecisionRequirementsApplier(state.getDmnDecisionState()));
```
Comment (korthout) [inline engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java:71]: 🔧 The DecisionRequirementsApplier should reflect the intent in its name as well as it only applies the state for the CREATED event.

**Review by korthout — APPROVED** [review]
Looks great 🏎️  Really clean commits and a good test suite along with it. 

Please have a look at my comments before merging, but definitely not blocking.

**Comment by saig0** [comment]
bors merge

**Comment by saig0** [comment]
bors merge

**Pull request merged**